### DISCUSSION TITLE
How to partially qualify class names

### DISCUSSION THREAD

**Discussion by chavan77me** [square/javapoet#1012] [Q&A]
```
class TopMost {
    class B {
    }
    class A {
        public static class B {
        }
        // I want this to extend A.B (but dont want it to fully qualify starting from the package name) 
        public static class C extends B {
        }
     }
 }
 ```
 
 How do I get JavaPoet to qualify B as A.B in the extends clause? I need to do this because there are other classes inside the outer class for A that also have the B name and C erroenously derives from that instead of B inside A
 
 Thank you for your help

**Comment by liujianpc** [comment]
您好，本人已收到您的邮件！谢谢！      祝您工作顺利，生活愉快！
